### PR TITLE
[13.0][MIG] account_invoice_line_sequence

### PR DESCRIPTION
**Pull request by thomaspaulb**
Non-standard migration from 12.0

Depends on:

- [ ] https://github.com/OCA/server-tools/pull/2306

### CODE REVIEW

**Review by HaraldPanten — CHANGES_REQUESTED** [review]
I'm afraid it's never going to be a standard migration because models have changed from V12 to V13

**Comment by thomaspaulb** [comment]
@HaraldPanten Good point, I found that out during initial testing :-) Changed it to non-standard migration

**Comment by HaraldPanten** [comment]
You should check pre-commit. Seems that black is failing.

**Comment by kittiu** [comment]
@thomaspaulb may be the travis error is from other module, `account_move_post_block`. 

Can you try to rebase to see if the error goes away? Thank you.

**Comment by thomaspaulb** [comment]
@HaraldPanten @kittiu Found one additional bug and fixed the test clash with `account_move_post_block`. If you have time, please review again.

@gjotten Maybe for your functional review as we are using for GW

**Review by AaronHForgeFlow — APPROVED** [review]
LGTM functional + code review

**Comment by navarrorico** [comment]
LGTM :+1:

**Comment by ofonseca-pyming** [comment]
@HaraldPanten 
Functional review, LGTM

**Review by ofonseca-pyming — APPROVED** [review]

**Comment by HaraldPanten** [comment]
@thomaspaulb could you rebase, please? I'd like to check the workflows 👍

**Comment by thomaspaulb** [comment]
@HaraldPanten Rebased.

**Comment by HaraldPanten** [comment]
Travis failure should be checked. Maybe @AaronHForgeFlow knows better than me how to solve the failing tests for account_invoice_triple_discount

**Comment by AaronHForgeFlow** [comment]
I am afraid do not know about that module, perhaps @CarlosRoca13 or @benwillig remember something.

File: account_invoice_line_sequence/views/account_invoice_view.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='invoice_line_ids']" position="before">
+                <field name="max_line_sequence" invisible="1" />
+            </xpath>
+            <xpath expr="//field[@name='invoice_line_ids']" position="attributes">
+                <attribute
+                    name="context"
+                    operation="python_dict"
+                    key="default_sequence"
+                >
```
Comment (thomaspaulb) [inline account_invoice_line_sequence/views/account_invoice_view.xml:16]: Found the issue, although not yet solved it.

This piece invokes code of `base_view_inheritance_extension`, but there is a bug: Odoo's [context.get('default_type')](https://github.com/OCA/OCB/blob/13.0/addons/account/views/account_move_views.xml#L674) is mistakenly seen as being [not a variable](https://github.com/OCA/server-tools/blob/13.0/base_view_inheritance_extension/models/ir_ui_view.py#L84) and as such is quoted.

Solution angles include:

- Patching `is_variable` function
- Backporting https://github.com/OCA/server-tools/pull/2216 and https://github.com/OCA/server-tools/pull/2305/files that reimplement this
- Doing this part without needing base_view_inheritance_extension (tips anyone?)

**Comment by thomaspaulb** [comment]
@AaronHForgeFlow @HaraldPanten Found the issue, see comment above - which route would you choose?

File: account_invoice_line_sequence/views/account_invoice_view.xml (hunk 1 above)
Comment (thomaspaulb) [reply account_invoice_line_sequence/views/account_invoice_view.xml:16]: I chose the second option and backported to https://github.com/OCA/server-tools/pull/2306

Comment (AaronHForgeFlow) [reply account_invoice_line_sequence/views/account_invoice_view.xml:16]: Hi @thomaspaulb , Thanks for the fix, I agree option 2 is the better one.

**Review by AaronHForgeFlow — APPROVED** [review]
Functional review working fine. Waiting for the fix in the unrelated third module.

File: account_invoice_line_sequence/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Provides a new sequence field on invoice lines which helps to manage the order of the invoice lines.
```
Comment (AaronHForgeFlow) [inline account_invoice_line_sequence/readme/DESCRIPTION.rst:1]: Can you improve the readme to mention that the sequence changes automatically when editing the lines and the sequence number is printed in the report?

**Comment by thomaspaulb** [comment]
@AaronHForgeFlow I updated the review and it's now green, after the other module got merged.

**Comment by AaronHForgeFlow** [comment]
For me this is fine. @HaraldPanten is this ok to you?

**Review by HaraldPanten — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
/ocabot merge nobump

Thank you!

**Pull request merged**